src/test/java/testsuite/integration/container/HikariCPIntegrationTest.java

```diff
@@ -0,0 +1,170 @@
+package testsuite.integration.container;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+import org.junit.jupiter.api.*;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLTransientConnectionException;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class HikariCPIntegrationTest extends AuroraMysqlIntegrationBaseTest {
+  private static HikariDataSource DATA_SOURCE = null;
+  private static final String jdbcUrl = "jdbc:mysql:aws://" + MYSQL_CLUSTER_URL + PROXIED_DOMAIN_NAME_SUFFIX + ":" + MYSQL_PROXY_PORT;
+
+  @BeforeAll
+  static void setup() throws IOException, SQLException {
+    AuroraMysqlIntegrationBaseTest.setUp();
+
+    System.setProperty("com.zaxxer.hikari.blockUntilFilled", "true");
+    HikariConfig config = new HikariConfig();
+
+    config.setJdbcUrl(jdbcUrl);
+    config.setUsername(TEST_USERNAME);
+    config.setPassword(TEST_PASSWORD);
+    config.setMaximumPoolSize(3);
+    config.setReadOnly(true);
+    config.setInitializationFailTimeout(75000);
+
+    DATA_SOURCE = new HikariDataSource(config);
+    System.out.println("Starting idle connections: " + DATA_SOURCE.getHikariPoolMXBean().getIdleConnections());
+    System.out.println("Starting active connections: " + DATA_SOURCE.getHikariPoolMXBean().getActiveConnections());
+    System.out.println("Starting total connections: " + DATA_SOURCE.getHikariPoolMXBean().getTotalConnections());
+  }
+
+  @AfterAll
+  static void teardown() {
+    DATA_SOURCE.close();
+  }
+
+  @BeforeEach
+  public void disconnectCluster() {
+    System.out.println("Putting down all instances before test");
+    putDownInstance(null);
```

Comment:
Can we have a function/method that says something about `putDownAll` instead of passing in `null`?